analyses/CLK1-splicing_correlations/08-plot-Ex4-PSI-all-tumors.R

```diff
@@ -0,0 +1,206 @@
+  # Join rmats data with clinical data
+  inner_join(histologies_df, by='Kids_First_Biospecimen_ID') %>%
+  mutate(gene_symbol="CLK1")
+
+exp <- readRDS(expr_file) %>%
+  select(any_of(histologies_df$Kids_First_Biospecimen_ID))
+
+var_exp_filt <- add_TPM_values(clk1_rmats, exp) 
+
+# filter for very high expression 
+high_expression <- quantile(var_exp_filt$gene_tpm, 0.90)
+ex4_psi_filtered <- var_exp_filt %>%
+   filter(gene_tpm > high_expression) %>%
+  # Compute variance and add as a new column
+  group_by(plot_group) %>%
+  mutate(PSI_variance = sd(IncLevel1, na.rm = TRUE)) %>%
+  filter(!is.na(PSI_variance))
+
+# Plot with pairwise comparison results and mean labels
+boxplot_tpm<- ggplot(var_exp_filt, aes(x = plot_group, y = IncLevel1)) +
+  geom_boxplot(outlier.shape = NA, fill = "grey", color = "#2c3e50") +
+  stat_summary(fun = mean, geom = "point", shape = 18, size = 3, color = "black") +
+  geom_jitter(width = 0.2, size = 2, shape = 21, color = "black", fill="grey") + # Add actual data points
+  labs(title = "CLK Exon 4 PSI", 
+       x = "Histology", y = "PSI") +
+  theme_Publication() + 
+  theme(legend.position = "none", 
+        axis.text.x = element_text(angle = 75, hjust = 1)) +
+  scale_x_discrete(labels = function(x) sapply(x, function(l) str_wrap(l, width = 30))) # Wrap x-axis labels 
+
+
+var_plot<- ggplot(data=ex4_psi_filtered, 
+       aes(reorder(plot_group, PSI_variance),PSI_variance,  
+           group=1), color="black") +
+  geom_point(aes(fill = 'black'), size = 3, pch = 21, color="black") +  # Map color inside aes()
+  xlab("Histology") + 
+  ylab("Standard Deviation") + 
+  ggtitle("CLK1 Exon 4 variation") +
+  theme_Publication() + 
+  theme(axis.text.x=element_text(angle = 75, hjust = 1, size = 11),legend.position = "none") 
+
+ex4_psi_range <- ex4_psi_filtered %>%
+  group_by(plot_group) %>%
+  mutate(PSI_range = max(IncLevel1, na.rm = TRUE) - min(IncLevel1, na.rm = TRUE)) %>%
+  ungroup() %>% 
+  select(plot_group,PSI_range) %>%
+  unique()
+
+psi_range_plot<- ggplot(data=ex4_psi_range, 
+       aes(reorder(plot_group, PSI_range),PSI_range,  
+           group=1), color="black") +
+  geom_point(aes(fill = 'black'), size = 3, pch = 21, color="black") +  # Map color inside aes()
+  xlab("Histology") + 
+  ylab("PSI Range") + 
+  ggtitle("CLK1 Exon 4 PSI Range") +
+  theme_Publication() + 
+  theme(axis.text.x=element_text(angle = 75, hjust = 1, size = 11),legend.position = "none") 
+
+
+# Save plot as PDF
+pdf(file.path(plots_dir, "CLK1-Ex4-sdev-across.pdf"), 
+    width = 4, height = 6)
+print(var_plot)
+dev.off()
+
+pdf(file.path(plots_dir, "CLK1-Ex4-range-across.pdf"), 
+    width = 4, height = 6)
+print(psi_range_plot)
+dev.off()
+
+## add ctrls
+## add ctrls
+gtex_rmats <- file.path("~/d3b_coding/neoepitope-identification/data/gtex-brain-under40-harmonized-splice-events-rmats.SE.tsv.gz")
+pedr_rmats <- file.path("~/d3b_coding/neoepitope-identification/data/GSE243682_normal_splice-events-rmats.tsv.gz")
```

Comment:
```suggestion
gtex_rmats <- file.path("~/d3b_coding/neoepitope-identification/data/gtex-brain-under40-harmonized-splice-events-rmats.SE.tsv.gz")
pedr_rmats <- file.path("~/d3b_coding/neoepitope-identification/data/GSE243682_normal_splice-events-rmats.tsv.gz")
```

do we need to put a new release together